### DISCUSSION TITLE
Import existing Database

### DISCUSSION THREAD

**Discussion by azevedonik** [directus/directus#10763] [Archive]
Hey guys. Is there a straightforward way to manage an existing DB? I tried creating a new table manually outside directus and it showed up inside the collections page. However, when I click to add an item I can add the item, but they don't show up after creating.

Database: 
![image](https://user-images.githubusercontent.com/14229409/147753803-7749abd7-fd18-4fc3-9299-9f769ac12b41.png)

Dashboard:
![image](https://user-images.githubusercontent.com/14229409/147753791-51420379-3fa6-4fca-9c87-4c43b9fe7879.png)

**Comment by azevedonik** [comment]
The error showing in the console:
![image](https://user-images.githubusercontent.com/14229409/147754698-a923d90c-04d7-4a12-943d-92cb4660b237.png)

**Reply by rijkvanzanten** [reply]
> Is there a straightforward way to manage an existing DB?

You can install Directus' tables on top of an existing database, or do exactly what you did. Do make sure [to flush the schema cache though](https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache) after a manual change to make sure Directus properly reads it (or turn it off completely if you expect to make a lot of changes in the DB directly)

**Reply by azevedonik** [reply]
Yes. I cleaned the cache. The problem is that when I setup the fields, directus is able to create items to the existing collection, however, it doesn't let me see the created items. I'm getting the error above in the console. Cannot read properties of null.

**Reply by rijkvanzanten** [reply]
Not sure! Would you be able to share a database dump with a minimal reproduction of this?

**Comment by azevedonik** [comment]
Sure ! 
Basically I did the following. Manually created a Table:
![image](https://user-images.githubusercontent.com/14229409/147795406-243612b8-b3db-41fa-9d07-5b0d32b5607f.png)

Then, restarted Directus and the table showed up.

Clicked to config:
![image](https://user-images.githubusercontent.com/14229409/147795437-d9ddbfd7-5cae-4e8b-b5a8-5a168fb9eb23.png)

Clicked on each field and configured accordingly.
The first problem is that Directus is not identifying id as the primary key:
![image](https://user-images.githubusercontent.com/14229409/147795475-bbab29bf-2321-4b4a-a049-0f8299eda7f2.png)

Then, I can go to the collection and click Create Item. However, after creating the item, I can not see anythin:
![image](https://user-images.githubusercontent.com/14229409/147795501-86c123e6-e50d-4e21-9ed4-49673f8bc2b2.png)

When inspecting the network response, the data is coming from the API, however, there's an error saying:
vendor.275c9c53.js:5 TypeError: Cannot read properties of null (reading 'field')
![image](https://user-images.githubusercontent.com/14229409/147795540-12a1bc80-9413-415b-a845-514820a09b0d.png)
Error when Opening the Collection added.
![image](https://user-images.githubusercontent.com/14229409/147795580-a9aef309-ba63-4c3f-8a49-bcf8dbe90086.png)

This only happens when I create a table manually.

DUMP FILE:

[dump.sql.zip](https://github.com/directus/directus/files/7794932/dump.sql.zip)

**Comment by azevedonik** [comment]
Actually after a long inspecting I realized the problem was that Datagrip created two indexes in the same field (ID)
![image](https://user-images.githubusercontent.com/14229409/147796307-4d5e2ef6-a02a-4720-941a-ef2248b91a0d.png)

After removing the second index:
![image](https://user-images.githubusercontent.com/14229409/147796319-bf19a0ca-c1fc-4799-a423-e8dc4128620d.png)

It detected:
![image](https://user-images.githubusercontent.com/14229409/147796321-a98ff4a4-5ec0-49e9-84cc-5c188013836b.png)


Is it a Bug ? What could be causing this? Any clue? After removing the second index, it just simply worked.

**Reply by rijkvanzanten** [reply]
Pretty sure that's the problem! It fails to recognize that field as the primary key of the table, hence it starts failing in the app

**Reply by azevedonik** [reply]
Hmm but, is it a bug from Datagrip or Directus? I don't see a point to have multiple indexes to the same field.
But thanks for all the support @rijkvanzanten

**Reply by rijkvanzanten** [reply]
Could be both! Directus should still recognize it as the primary key though!
